cellmlmanip/model.py

```diff
@@ -711,11 +769,10 @@ def convert_variable(self, original_variable, units, direction):
         # if free variable
         if original_variable == free_symbol:
             # for each derivative wrt to free variable add necessary variables/equations
-            current_equations = self.equations.copy()
-            for equation in current_equations:
-                if equation.args[0].is_Derivative:
-                    if equation.args[0].args[1].args[0] == original_variable:
-                        new_derivatives.append(self._convert_free_variable_deriv(equation, new_variable, cf))
+            for ode in [eq for v, eq in sorted(self._ode_definition_map.items(),
+                                               key=lambda v_eq: v_eq[0].order_added)]:
```

Comment:
Without it you get the equations list in an unpredictable order on Python 3.5, leading to occasional test failures. Because `_convert_free_variable_deriv` adds equations to the model.